src/script/interpreter.cpp

```diff
@@ -1039,6 +1043,20 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
     if (!vfExec.empty())
         return set_error(serror, SCRIPT_ERR_UNBALANCED_CONDITIONAL);
 
+    if (allow_tail_call && !stack.empty() && ((stack.size() + altstack.size()) >= 2) && CastToBool(stack.back())) {
+        // Replace the script we just finished executing with the
+        // subscript from the top of the stack:
+        const valtype& policyScript = stacktop(-1);
+        script = CScript(policyScript.begin(), policyScript.end());
+        popstack(stack);
+        // Only allow one tail-call:
+        allow_tail_call = false;
```

Comment:
Apologies for being annoying//this may be a stupid question: but why only do TCO once?

Reply:
https://github.com/bitcoin/bips/blob/master/bip-0117.mediawiki#rationale

Reply:
Thank you very much!